woocommerce/woocommerce-ios
# [Mobile Payments] Add tap to pay setup screen

Part of: #9010


## Description


In M2 of sTAP Away, we are adding a set up screen for Tap to Pay which we can direct merchants to, in order that they can do the longer-running initial set up of TTP, then try out a payment.

This PR adds the new screen, complete with the set up action which initiates the connection process.

The flow is incomplete, but that's OK because this is behind a feature flag. When the connection finishes, the merchant is currently taken to a blank screen – this will in future be a "Connected" celebration screen, with a path to try out a payment.

In a follow-up PR, I'll add the cancel button and some tests, but this one is getting a little long.

## Testing instructions


Using an iPhone XS or newer on iOS 16+, with a US-based WCPay/Stripe store, and a local Xcode build:

1. Navigate to `Menu > Payments`
2. Wait for `Manage card reader` to be enabled (resolve any onboarding issues if required)
3. Tap the `Set up Tap to Pay on iPhone` row
4. Observe that you see the new screen
5. Tap the `Set up Tap to Pay on iPhone` button
6. Observe that the connection process runs, as it does for a payment, but no payment flow is started.
7. Observe that when complete, you see a blank screen.

You can also test with a store that's not yet accepted Apple's ToS, and during the connection process you will see the ToS acceptance flow.

### Known issue
Tapping the new row before `Manage card reader` is enabled will do nothing. In a future PR, this will either be disabled following the same rules as for `Manage card reader`, or show the onboarding steps, as yet undecided.


## Screenshots



https://user-images.githubusercontent.com/2472348/222758717-5bd23cda-9700-4f90-987b-76fd46c1c7e5.mp4


---
